test/distributed/checkpoint/test_state_dict.py

```diff
@@ -58,7 +58,7 @@ class TestStateDict(DTensorTestBase, VerifyStateDictMixin):
 
     @property
     def world_size(self) -> int:
-        return 2
+        return min(torch.cuda.device_count(), 2)
```

Comment:
What if `device_count()` returns 0?
What if someone want to run this test on non-cuda devices?

Reply:
All our tests have the skip_if statements which will ignore the tests if the gpu count is less than some value. The issue is that world_size() is executed before that.

Reply:
Actually we no longer see error with test from test_state_dict.py. So removing the change to this file.